tests/e2e/flowtest/test_maintenance_pod.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import time
+from ocs_ci.ocs import constants
+from ocs_ci.framework.pytest_customization.marks import (
+    skipif_ocs_version,
+    bugzilla,
+    tier2,
+    e2e,
+)
+from ocs_ci.framework.testlib import E2ETest
+from ocs_ci.ocs.resources.deployment import (
+    get_osd_deployments,
+    get_deployments_having_label,
+    get_mon_deployments,
+)
+from ocs_ci.ocs.resources.pod import (
+    get_pods_having_label,
+    Pod,
+    wait_for_pods_to_be_running,
+)
+from ocs_ci.utility.utils import ceph_health_check
+
+logger = logging.getLogger(__name__)
+
+
+@tier2
+@e2e
+@bugzilla("2103256")
+@skipif_ocs_version("<4.12")
+class TestMaintenancePod(E2ETest):
+    def test_maintenance_pod_for_osd(self, ceph_objectstore_factory):
+        """
+        Test Maintenance Pod for OSD
+        """
+        label = "ceph-osd-id=0,ceph.rook.io/do-not-reconcile=true"
+        original_deployment = "rook-ceph-osd-0"
+
+        # enable the debug mode for osd
+        Cot_obj = ceph_objectstore_factory
+        Cot_obj.debug_start(deployment_name=original_deployment)
+
+        # make sure original deployment is scaled down
+        # make sure the new debug pod is brought up and running successfully
+        osd_deployments = get_osd_deployments()
+        for deployment in osd_deployments:
+            if deployment.name == original_deployment and deployment.replicas != 0:
+                raise Exception(
+                    f"Original deployment {original_deployment} is not scaled down!"
+                )
+
+        debug_deployment = get_deployments_having_label(
+            label=label, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        if len(debug_deployment) == 0:
+            raise Exception("Debug deployment is not up!")
+        wait_for_pods_to_be_running(pod_names=[debug_deployment[0].pods[0].name])
+        logger.info("Verified debug deployment is up & running!")
+
+        # Run any COT operations
+        time.sleep(5)
```

Comment:
why do we need sleep here?

Reply:
sometime even though debug pods are up & running. it takes time for the container to initialize. This can lead to COT operations to fail